File: web/src/api/jxc/sale.js
import service from '@/utils/request'

// 销售单
export const getSalePage = (params) => service.get('/jxc/sale/page', { params })
export const getSaleDetail = (id) => service.get('/jxc/sale/detail', { params: { id } })
export const createSaleOrder = (data) => service.post('/jxc/sale', data)
export const updateSaleOrder = (data) => service.put('/jxc/sale', data)
export const deleteSaleOrder = (id) => service.delete('/jxc/sale', { data: { id } })
export const cancelSaleOrder = (id) => service.put('/jxc/sale/cancel', { id })
export const confirmOut = (id) => service.put('/jxc/sale/out', { id })
export const confirmReturn = (id) => service.put('/jxc/sale/return', { id })
export const confirmExchange = (id) => service.put('/jxc/sale/exchange', { id })

File: server/router/jxc/sale.go
package jxc

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type SaleRouter struct{}

func (r *SaleRouter) InitSaleRouter(Router *gin.RouterGroup) {
	recordGroup := Router.Group("jxc").Use(middleware.OperationRecord())
	readOnlyGroup := Router.Group("jxc")

	// ===== 销售单 =====
	readOnlyGroup.GET("sale/page", saleApi.GetSalePage)
	readOnlyGroup.GET("sale/detail", saleApi.GetSaleDetail)
	readOnlyGroup.GET("sale/remaining", saleApi.GetSaleRemaining)
	recordGroup.POST("sale", saleApi.CreateSaleOrder)
	recordGroup.PUT("sale", saleApi.UpdateSaleOrder)
	recordGroup.DELETE("sale", saleApi.DeleteSaleOrder)
	recordGroup.PUT("sale/cancel", saleApi.CancelSaleOrder)
	recordGroup.PUT("sale/out", saleApi.ConfirmOut)
	recordGroup.PUT("sale/return", saleApi.ConfirmReturn)
	recordGroup.PUT("sale/exchange", saleApi.ConfirmExchange)
}
